static/js/components/forms/SiteContentForm.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from "react"
+import { partition } from "ramda"
+import { Form, Formik, FormikHelpers } from "formik"
+
+import SiteContentField from "./SiteContentField"
+import {
+  contentInitialValues,
+  fieldIsVisible,
+  isMainContentField,
+  newInitialValues
+} from "../../lib/site_content"
+import { getContentSchema } from "./validation"
+
+import {
+  ConfigField,
+  EditableConfigItem,
+  WebsiteContent
+} from "../../types/websites"
+import { ContentFormType } from "../../types/forms"
+
+export type SiteFormValues = {
+  [key: string]: string
+}
+
+export const splitFieldsIntoColumns = (
+  fields: ConfigField[]
+): ConfigField[][] =>
+  partition(isMainContentField, fields).filter(column => column.length > 0)
+
+type Props = {
+  onSubmit: (
+    values: any,
+    formikHelpers: FormikHelpers<any>
+  ) => void | Promise<any>
+  configItem: EditableConfigItem
+  content: WebsiteContent
+  formType: ContentFormType
+}
+
+export default function SiteContentForm({
+  onSubmit,
+  configItem,
+  content,
+  formType
+}: Props): JSX.Element {
+  const fields = configItem.fields
+  const initialValues: SiteFormValues =
+    formType === ContentFormType.Add ?
+      newInitialValues(fields) :
+      contentInitialValues(content, fields)
+  const schema = getContentSchema(configItem)
+
+  const fieldsByColumn = splitFieldsIntoColumns(fields)
+  const columnClass = fieldsByColumn.length === 2 ? "col-6" : "col-12"
+  const buttonColumnIndex = fieldsByColumn.length - 1
+
+  return (
+    <Formik
+      onSubmit={onSubmit}
+      validationSchema={schema}
+      initialValues={initialValues}
+    >
+      {({ isSubmitting, status, setFieldValue, values }) => (
+        <Form className="row">
+          {fieldsByColumn.map((columnFields, idx) => (
+            <div className={columnClass} key={idx}>
+              {columnFields.map(field =>
+                fieldIsVisible(field, values) ? (
+                  <SiteContentField
+                    field={field}
+                    key={field.name}
+                    setFieldValue={setFieldValue}
+                  />
+                ) : null
+              )}
+              {buttonColumnIndex === idx ? (
+                <>
+                  <div className="form-group d-flex justify-content-end">
+                    <button
+                      type="submit"
+                      disabled={isSubmitting}
+                      className="px-5 btn blue-button"
+                    >
+                      Save
+                    </button>
+                  </div>
```

Comment:
why don't we just render these buttons and the `.form-error` status below in a separate full-width div after we've closed out the columns? it seems a bit confusing that this is in the `fieldsByColumn.map` call